overlay_gl/overlay.c

```diff
@@ -147,25 +147,37 @@ static void newContext(Context *ctx) {
 	ctx->timeT             = clock();
 	ctx->frameCount        = 0;
 
+#ifdef __linux__
+	char *xdgRuntimeDir = getenv("XDG_RUNTIME_DIR");
+
+	if (xdgRuntimeDir != NULL) {
+		ctx->saName.sun_family = PF_UNIX;
+		strcpy(ctx->saName.sun_path, xdgRuntimeDir);
+		if(xdgRuntimeDir[(strlen(xdgRuntimeDir)-1)] != '/')
+			strcat(ctx->saName.sun_path, "/");
+		strcat(ctx->saName.sun_path, "mumble/MumbleOverlayPipe");
+	} else {
+		char uid[10];
+		sprintf(uid, "%d", getuid());
+		ctx->saName.sun_family = PF_UNIX;
+		strcpy(ctx->saName.sun_path, "/run/user/");
+		strcat(ctx->saName.sun_path, uid);
+		strcat(ctx->saName.sun_path, "/mumble/MumbleOverlayPipe");
+	}
+#else
```

Comment:
All of these ifdef clauses should instead simply call the same function that we are using in the mumble client. I think at this point the only way to achieve this, would be to copy the respective file(s) from the main Mumble source tree via cmake and add those as source files for this target. Shouldn't cause any issues though and also avoids us having to keep duplicated code around.